pkg/controllers/metrics/metrics.go

```diff
@@ -0,0 +1,77 @@
+package metrics
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"strconv"
+)
+
+var (
+	totalRequests = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "custom_metric_http_requests_total",
+			Help: "Number of get requests.",
+		}, []string{"path"},
+	)
+
+	responseStatus = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "custom_metric_response_status",
+			Help: "Status of HTTP response",
+		},
+		[]string{"status"},
+	)
+
+	httpDuration = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "custom_metric_http_response_time_seconds",
+		Help:    "Duration of HTTP requests.",
+		Buckets: prometheus.LinearBuckets(0.001, 0.003, 10),
+	}, []string{"path"})
+
+	metricList = []prometheus.Collector{
+		totalRequests,
+		responseStatus,
+		httpDuration,
+	}
+)
+
+func NewMetricsController(router *gin.Engine) {
+	RegisterCustomMetrics()
+	router.Use(MiddlewareMetrics())
+	router.GET("/metrics", HandlerMetrics())
+}
+
+func RegisterCustomMetrics() {
+	for _, metric := range metricList {
+		err := prometheus.Register(metric)
+		if err != nil {
```

Comment:
It's better not to ignore errors. One day it could be a bug that is hard to find. Please log it.

Reply:
Got it. Added error logging to this function in the second commit.